app.py

```diff
@@ -44,4 +44,176 @@ def show_breakdown_dialog(self):
+         'sg_data': {'code': 'aaa_00010_test_output_v001.%04d.dpx',
+                     'entity': {'id': 1660, 'name': 'aaa_00010', 'type': 'Shot'},
+                     'id': 1424,
+                     'name': 'test',
+                     'published_file_type': {'id': 3,
+                                             'name': 'Rendered Image',
+                                             'type': 'PublishedFileType'},
+                     'task': {'id': 4714, 'name': 'Comp', 'type': 'Task'},
+                     'type': 'PublishedFile',
+                     'project': {'id': 234, 'name': 'Climp', 'type': 'Project'},
+                     'version_number': 1},
+         'template': <Sgtk TemplatePath nuke_shot_render_pub_mono_dpx>}
+        
+        This method will attempt to connect to shotgun, but the number of calls made are
+        constant and independent of the scene complexity.
+        
+        You typically use it like this:
+        
+        >>> import sgtk
+        >>> e = sgtk.platform.current_engine()
+        >>> items = e.apps["tk-multi-breakdown"].analyze_scene()
+        
+        :returns: List of dictionaries, see above for example.
+        """
+        tk_multi_breakdown = self.import_module("tk_multi_breakdown")
+        
+        # first, scan the scene and get a list of items
+        items = tk_multi_breakdown.get_breakdown_items()
+        
+        # if shotgun data is returned for an item, trim this down
+        # to return a more basic listing than the one returned
+        # from get_breakdown_items:
+        for i in items:
+            if i["sg_data"]:
+                new_sg_data = {}
+                new_sg_data["id"] = i["sg_data"]["id"]
+                new_sg_data["type"] = i["sg_data"]["type"]
+                new_sg_data["code"] = i["sg_data"]["code"]
+                new_sg_data["entity"] = i["sg_data"]["entity"]
+                new_sg_data["task"] = i["sg_data"]["task"]
+                new_sg_data["published_file_type"] = i["sg_data"]["published_file_type"]
+                new_sg_data["name"] = i["sg_data"]["name"]
+                new_sg_data["project"] = i["sg_data"]["project"]
+                new_sg_data["version_number"] = i["sg_data"]["version_number"]
+                i["sg_data"] = new_sg_data
+                
+        return items
+
+
+    def calculate_latest_version(self, template, fields):
+        """
+        Given a template and some fields, return the highest version number found on disk.
+        The template key containing the version number is assumed to be named {version}.
+        
+        This will perform a scan on disk to determine the highest version.
         
+        You typically use this in conjunction with the analyze_scene() method. 
+        For example:
+        
+        import sgtk
+        e = sgtk.platform.current_engine()
```

Comment:
is 'e' used anywhere else (it's pretty hard to search for as it's a single character!) - if it's not then I'd be inclined to collapse these two lines into one and get rid of 'e', otherwise use more letters (e.g. 'eng') :)